librz/bin/p/bin_sms.c

```diff
@@ -12,29 +12,64 @@ typedef struct gen_hdr {
 	ut8 RegionRomSize; // Low 4 bits RomSize, Top 4 bits Region
 } SMS_Header;
 
-static ut32 cb = 0;
+typedef struct sms_info {
+	SMS_Header *hdr;
+	ut32 offset;
+} SMS_INFO;
 
-static bool check_buffer(RzBuffer *b) {
+static bool find_magic(RzBuffer *b, ut32 *offset) {
 	ut32 *off, offs[] = { 0x2000, 0x4000, 0x8000, 0x9000, 0 };
 	ut8 signature[8];
 	for (off = (ut32 *)&offs; *off; off++) {
 		rz_buf_read_at(b, *off - 16, (ut8 *)&signature, 8);
 		if (!strncmp((const char *)signature, "TMR SEGA", 8)) {
-			cb = *off - 16;
+			if (offset) {
```

Comment:
Check `offset` for `NULL` once at the beginning of the function and return false if not set. It doesn't make sense otherwise in the bigger picture.